### ISSUE TITLE
Executing the same rollout scripts twice result in error when executing the RBAC file

### ISSUE THREAD

**Issue by dannystommen** [Snowflake-Labs/roleout#6]
We have the following yaml file:
`
project:
  name: Groupm Snowflake
  environments:
    - name: DEV
  functionalRoles:
    - name: DEVELOPER_READONLY
  databases:
    - name: DANNY
      schemata:
        - name: JARVIS
          access:
            - role: DEVELOPER_READONLY
              level: Read
              env: DEV
          options:
            - env: DEV
              managedAccess: true
              transient: false
      options:
        - env: DEV
          transient: false
  virtualWarehouses:
    - name: TEST
      options:
        - env: DEV
          size: MEDIUM
          minClusterCount: 1
          maxClusterCount: 1
          autoSuspend: 10
          autoResume: true
          scalingPolicy: STANDARD
      access: []

`

When creating the SQL scripts u
[roleout_scripts.zip](https://github.com/Snowflake-Labs/roleout/files/11636181/roleout_scripts.zip)
sing the roleout cli
`
roleout sql deploy -c roleout.yml -o roleout_scripts
`
It creates the attached SQL files.

When executing these scripts twice, it fails on the 2nd run on the `04 - RBAC.sql` file. It fails on the first GRANT statement 
`
GRANT OWNERSHIP ON SCHEMA "DEV_DANNY_DB"."JARVIS" TO ROLE "DEV_DANNY_DB_JARVIS_FULL_AR" REVOKE CURRENT GRANTS;
`
with the following error
`
SQL execution error: One or more ownership future grants exist in the schema. They must be revoked first.
`

**Comment by sfc-gh-ckite** [comment]
As of now that is the expected behavior. In plain SQL we don't have a way to know which future grants already exist in a schema, and Snowflake currently does not have the ability to revoke all future grants when transferring ownership.

The recommended workflow when using SQL in Roleout is to use the `teardown` scripts to remove previous RBAC, and then run the `deploy` script to put the new RBAC structure in place. See the [SQL Limitations section of the README](https://github.com/Snowflake-Labs/roleout#sql-limitations) for details.

**Comment by dannystommen** [comment]
@sfc-gh-ckite Thanks for explaining. Would using the Terraform variant solve this issue?

**Comment by sfc-gh-ckite** [comment]
@dannystommen definitely. Terraform is able to track changes over time and keep your account up to date with your Terraform resources created by Roleout.

**Comment by dannystommen** [comment]
@sfc-gh-ckite Thanks! Will try it out somewhere soon!
